scanner/scanner.go

```diff
@@ -145,35 +150,35 @@ func (s *Scanner) processMatcherEntry(matcher Matcher, info entryInfo, foundCert
 	return nil
 }
 
-func (s *Scanner) processMatcherLeafEntry(matcher LeafMatcher, info entryInfo, foundCert func(*ct.LogEntry), foundPrecert func(*ct.LogEntry)) error {
+func (s *Scanner) processMatcherLeafEntry(matcher LeafMatcher, info entryInfo, foundCert func(*ct.RawLogEntry), foundPrecert func(*ct.RawLogEntry)) error {
 	if !matcher.Matches(&info.entry) {
 		return nil
 	}
 
-	logEntry, err := ct.LogEntryFromLeaf(info.index, &info.entry)
-	if logEntry == nil {
-		return fmt.Errorf("failed to build log entry %d: %v", info.index, err)
+	rawLogEntry, err := ct.RawLogEntryFromLeaf(&info.entry, info.index)
+	if rawLogEntry == nil {
+		return fmt.Errorf("failed to build raw log entry %d: %v", info.index, err)
 	}
-	switch {
-	case logEntry.X509Cert != nil:
+	switch rawLogEntry.Leaf.TimestampedEntry.EntryType {
```

Comment:
nit:
`switch eType := rawLogEntry.Leaf.TimestampedEntry.EntryType; eType {`
... and use `eType` in line 173 for error message

Reply:
Done.